Rules rbac siemv5 role update

denar50 opened the pull request:
## 🚧 TO BE MERGED AFTER  #239634 🚧 

See [HERE](https://github.com/rylnd/kibana/compare/rules-rbac-new...denar50:rules-rbac-siemv5-role-update) for a diff _without_ all of the "base" RBAC changes.

This updates prebuilt roles and those used in tests to:

* Reference siemv5 instead of the older siemv4
* Add the new `rules:read` or `rules:all` feature where appropriate

Without this change, our tests are implicitly testing the
`replacedBy`/"migration" path that existing users will follow. With that
version of the code being green, we can have confidence in the behavior
for existing users, and can then update our tests to use the latest
features, here.

rylnd committed:
Introduces Rules feature and subsequent app changes

This does not include changes to existing roles, nor the role migration
machinery.

rylnd committed:
style: revert linting changes to large yaml/JSON files

These changes were made automatically in an initial commit that added
our new features to roles; those changes have since been reverted
(320c34f485), and thus there should not currently be any behavioral
changes in these files, which makes these stylistic changes even more
unnecessary.

Note: I also noticed that a few old references had (accidentally?)
remained in `security_roles.json` after `320c34f485`; this cleans those
up as well.

denar50 committed:
update the siem migrations required permissions

Instead of requiring siemVX read/all, it now requires securitySolutionRulesV1 read/all

denar50 committed:
update attack discovery required permissions

denar50 committed:
update timeline required permissions

denar50 committed:
fix onboarding sections

It is unclear on wether "dashboards" and "integrations" should be exclusive to `siemV5` or `securitySolutionRulesV1`. So for now we are showing it when the user has either of those.

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

denar50 committed:
allow alert management operations for roles with read only access

This is the current behaviour accoring do the documentation https://www.elastic.co/docs/solutions/security/detect-and-alert/detections-requirements.

denar50 committed:
fix create shared exception list endpoint

Now it requires the `securitySolutionRulesV1.all` privilege

denar50 committed:
fix unit test

dplumlee committed:
fixes bulk_action route to allow export on rules-read

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

denar50 committed:
fix cypress test: endpoint_role_rbac_with_space_awareness.cy.ts

No security subfeature is required in all spaces anymore. The test was failing because the `siemV5` feature file never got updated and it was still referencing a feature flag that has been enabled and removed in `main`.
The feature flag in question is `endpointManagementSpaceAwarenessEnabled` which was being used to override the subfeature configuration by setting `requireAllSpaces=false` and `privilegesTooltip=undefined`. Now that the feature flag doesn't exist, it makes sense to remove these properties directly in the subfeature configuration instead of overriding them outside of it.

denar50 committed:
Merge branch 'main' into rules-rbac-new

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

denar50 committed:
fix manage value lists button disabled for rules-all

The logic to show it was relying on the old siemPrivileges, however value lists is now under rules.

dplumlee committed:
fix showing rule update callouts when users don't have rules-all

denar50 committed:
fix authorization tests

Reshuffling privileges and removal of alerting privileges from siemV5. These alerting privileges exist exclusively in securitySolutionRulesV1

denar50 committed:
Merge branch 'main' into rules-rbac-new

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

denar50 commented:
/ci

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/357282)
* Commit: dab712adecc29d7b43b6f4cc734c0c182fd6a651

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/357282#019a3b10-bd32-447b-b3e0-634a27091206)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357282","number":357282,"commit":"dab712adecc29d7b43b6f4cc734c0c182fd6a651"}],"number":357282}
buildkite-pr-comment-->

rylnd committed:
Merge branch 'main' into rules-rbac-new

This notably includes the fix to the infinite loop on the alerts page
when a role lacks sufficient lists privileges.

denar50 committed:
Merge remote-tracking branch 'upstream/main' into rules-rbac-1

denar50 committed:
add missing socManagement sub feature config to siemV4 and siemV5

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/357656)
* Commit: 9d59f5c9680acda467ea5a8ab1044c62399753d5

### Failed CI Steps
* [Defend Workflows Cypress Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2d5-4ffd-be40-7ad258b7fa78)
* [Defend Workflows Cypress Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2d5-45af-ad2b-e16bc47c8d71)
* [Defend Workflows Cypress Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2d6-4737-8267-eaa26be38338)
* [Defend Workflows Cypress Tests #5](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2d6-48fa-9e0d-a814d97a748d)
* [Defend Workflows Cypress Tests #6](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2d7-40c4-9a38-88a55390eb20)
* [Defend Workflows Cypress Tests #7](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2d7-4af6-b892-43e9245588d4)
* [Defend Workflows Cypress Tests #8](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2d8-42c9-992f-17d2dd28af6d)
* [Defend Workflows Cypress Tests #9](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2d8-4fe5-bf8f-4f0cfaaab260)
* [Defend Workflows Cypress Tests #10](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2d8-4a3a-b999-ed1c6fb0ed55)
* [Defend Workflows Cypress Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2d9-4d2a-8f60-8cd83f262ee9)
* [Defend Workflows Cypress Tests #12](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2d9-4150-a2b4-e9121c9730fc)
* [Defend Workflows Cypress Tests #13](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2d9-42ea-9c7d-e387b5a774e6)
* [Defend Workflows Cypress Tests #14](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2da-405e-af4e-3f10b4cdecfa)
* [Defend Workflows Cypress Tests #15](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2da-446c-ad3f-4151693dec9d)
* [Defend Workflows Cypress Tests #16](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2db-4573-9f6b-942029234b0b)
* [Defend Workflows Cypress Tests #17](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2db-4c2e-a4f1-d4dc8293a975)
* [Defend Workflows Cypress Tests #18](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2db-4c3b-a4db-67b843eafbc8)
* [Defend Workflows Cypress Tests #19](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2dc-49da-8765-430f2420eebb)
* [Defend Workflows Cypress Tests #20](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2dc-4fd0-98f1-525eb5eb4ffe)
* [Defend Workflows Cypress Tests on Serverless #1](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2dd-40b5-aed3-2c09ee35c689)
* [Defend Workflows Cypress Tests on Serverless #2](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2dd-4c22-ba71-aa91ab4cefe7)
* [Defend Workflows Cypress Tests on Serverless #3](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2de-4197-8cc6-92ed99c3e5c9)
* [Defend Workflows Cypress Tests on Serverless #4](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2df-406a-b9e6-bb5125f256b9)
* [Defend Workflows Cypress Tests on Serverless #5](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2df-4b1d-84b2-656a8fa5b37a)
* [Defend Workflows Cypress Tests on Serverless #6](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2df-4bd6-a31c-a3d2a73d355a)
* [Defend Workflows Cypress Tests on Serverless #7](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2e0-45af-9cb9-cf701502a333)
* [Defend Workflows Cypress Tests on Serverless #8](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2e0-4349-b69c-05a4a19265d9)
* [Defend Workflows Cypress Tests on Serverless #9](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2e0-4dfd-bf3f-777b00bd0cb4)
* [Defend Workflows Cypress Tests on Serverless #10](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2e1-4fbc-a8de-2935387a2be8)
* [Defend Workflows Cypress Tests on Serverless #11](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2e1-4564-997d-d8dfa25d671c)
* [Defend Workflows Cypress Tests on Serverless #12](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2e1-4437-9bb8-0a66027b97ee)
* [Defend Workflows Cypress Tests on Serverless #13](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2e2-414e-8b8f-2c708b9e3ba4)
* [Defend Workflows Cypress Tests on Serverless #14](https://buildkite.com/elastic/kibana-pull-request/builds/357656#019a49bd-e2e2-4d2c-8ba6-2fce31a96e78)

### History
* :broken_heart: [Build #357282](https://buildkite.com/elastic/kibana-pull-request/builds/357282) failed dab712adecc29d7b43b6f4cc734c0c182fd6a651

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357656","number":357656,"commit":"9d59f5c9680acda467ea5a8ab1044c62399753d5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357282","number":357282,"commit":"dab712adecc29d7b43b6f4cc734c0c182fd6a651"}],"number":357656}
buildkite-pr-comment-->

denar50 committed:
fix ai4soc cypress test
The test broke after merging main into the branch

denar50 committed:
add missing siemV5 case to trusted devices rbac cypress test

denar50 committed:
fix authorization tests after new ai4soc changes

denar50 committed:
fix api privileges tests after ai4soc changes

dplumlee committed:
fixes respectLicenseLevel test that broke merging upstream

denar50 committed:
Merge branch 'main' into rules-rbac-new

denar50 committed:
add cypress tests for the rules management page

dplumlee committed:
removes unused code and resolves PR TODOs

denar50 committed:
Updates roles/tests/test roles to use new Features

This updates prebuilt roles and those used in tests to:

* Reference siemv5 instead of the older siemv4
* Add the new `rules:read` or `rules:all` feature where appropriate

Without this change, our tests are implicitly testing the
`replacedBy`/"migration" path that existing users will follow. With that
version of the code being green, we can have confidence in the behavior
for existing users, and can then update our tests to use the latest
features, here.

denar50 commented:
/ci

denar50 committed:
update SECURITY_FEATURE_ID constant to siemV5

denar50 committed:
Merge branch 'main' into rules-rbac-new

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/358473)
* Commit: 9dbe25365972c24f1e79528350d2cc4397065cf7
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-241482)

### Failed CI Steps
* [Rule Management - Security Solution Cypress Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/358473#019a5389-3cf5-4058-8a0e-34589c72d5fb)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358473#019a5389-3cf5-4058-8a0e-34589c72d5fb) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358473/jobs/019a5389-3cf5-4058-8a0e-34589c72d5fb/artifacts/019a53d1-5ea3-4c55-836f-d520113702e3) Rule Management - Security Solution Cypress Tests #<span></span>3 / Rules table - privileges securitySolutionRulesV1 none User sec_all_user with role(s) sec_all_role should not be able to see the rules management page User sec_all_user with role(s) sec_all_role should not be able to see the rules management page

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358473","number":358473,"commit":"9dbe25365972c24f1e79528350d2cc4397065cf7"}],"number":358473}
buildkite-pr-comment-->

denar50 committed:
fix rules management page privileges cypress test failure

dplumlee committed:
removes remaining TODOs and unskips tests

denar50 committed:
add cypress tests for the alerts page

denar50 committed:
Merge branch 'main' into rules-rbac-new

denar50 committed:
add cypress tests for the rule details page

dplumlee committed:
Merge branch 'main' into rules-rbac-new

dplumlee committed:
switches detection engine health setup endpiont to have initalize-sec-sol permission

dplumlee committed:
Merge remote-tracking branch 'upstream/main' into rules-rbac-new

dplumlee committed:
removes commented out code

dplumlee committed:
fixes attack page merge conflicts

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/361591)
* Commit: 078d05638cd6fc9b7a4d6b29b1d062494769fa1a
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-241482)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/6e0be3e139e9bca1cf48e72a2789cda94c8d515d) | [after](https://github.com/elastic/kibana/commit/078d05638cd6fc9b7a4d6b29b1d062494769fa1a) | diff |
| --- | --- | --- | --- |
| `@kbn/security-solution-features` | 30 | 31 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/6e0be3e139e9bca1cf48e72a2789cda94c8d515d) | [after](https://github.com/elastic/kibana/commit/078d05638cd6fc9b7a4d6b29b1d062494769fa1a) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 11.1MB | 11.1MB | -1.2KB |

#### Public APIs missing exports
> Total count of every type that is part of your API that should be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. `Run node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/6e0be3e139e9bca1cf48e72a2789cda94c8d515d) | [after](https://github.com/elastic/kibana/commit/078d05638cd6fc9b7a4d6b29b1d062494769fa1a) | diff |
| --- | --- | --- | --- |
| `@kbn/security-solution-features` | 10 | 11 | +1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/6e0be3e139e9bca1cf48e72a2789cda94c8d515d) | [after](https://github.com/elastic/kibana/commit/078d05638cd6fc9b7a4d6b29b1d062494769fa1a) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 107.1KB | 107.7KB | +526.0B |
| `securitySolutionEss` | 40.1KB | 40.2KB | +61.0B |
| `securitySolutionServerless` | 56.8KB | 56.9KB | +55.0B |
| total |  |  | +642.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/6e0be3e139e9bca1cf48e72a2789cda94c8d515d) | [after](https://github.com/elastic/kibana/commit/078d05638cd6fc9b7a4d6b29b1d062494769fa1a) | diff |
| --- | --- | --- | --- |
| `@kbn/security-solution-features` | 36 | 37 | +1 |


</details>


### History
* :yellow_heart: [Build #359528](https://buildkite.com/elastic/kibana-pull-request/builds/359528) was flaky 9297b57442c1870816b9f25ae3557b0710a30663
* :broken_heart: [Build #358473](https://buildkite.com/elastic/kibana-pull-request/builds/358473) failed 9dbe25365972c24f1e79528350d2cc4397065cf7

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361591","number":361591,"commit":"078d05638cd6fc9b7a4d6b29b1d062494769fa1a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359528","number":359528,"commit":"9297b57442c1870816b9f25ae3557b0710a30663"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358473","number":358473,"commit":"9dbe25365972c24f1e79528350d2cc4397065cf7"}],"number":361591}
buildkite-pr-comment-->

rylnd committed:
Endpoint Operations Analyst can only read rules.

Reference:
https://docs.google.com/spreadsheets/d/16aGow187AunLCBFZLlbVyS81iQNuMpNxd96LOerWj4c
